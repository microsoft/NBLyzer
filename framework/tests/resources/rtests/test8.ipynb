{"cells":[{"cell_type":"code","execution_count":null,"metadata":{"_cell_guid":"49bdb088-6105-4191-b5c5-305b753ba7a9","_uuid":"54b1eadb99f669def7844fb4ae4d5df338fca94c"},"outputs":[],"source":["import numpy as np #  algebra\n","import pandas as pd # data processing\n","import matplotlib.pyplot as plt #Provides MATLAB-like plotting framework\n","import seaborn as sns #python visualization library\n","from sklearn.neighbors import NearestNeighbors\n","color = sns.color_palette()\n","\n","train = pd.read_csv('../input/train_2016_v2.csv' , parse_dates=[\"transactiondate\"])\n","prop = pd.read_csv('../input/properties_2016.csv', usecols=['longitude', 'latitude', 'parcelid', 'regionidzip',\n","                                                            'propertyzoningdesc', 'taxvaluedollarcnt', 'structuretaxvaluedollarcnt'])"]},{"cell_type":"code","execution_count":null,"metadata":{"_cell_guid":"5d1dc9a2-55ac-4ea1-9435-7c2544411a01","_uuid":"3707e736a3c20e51af93e18a141f84e4f0bf2c76","collapsed":true},"outputs":[],"source":["train = pd.merge(train, prop, on=['parcelid'], how='left')\n","train.sort_values('transactiondate', inplace=True)\n","train['latitude'] = train['latitude']/1e6\n","train['longitude'] = train['longitude']/1e6"]},{"cell_type":"code","execution_count":null,"metadata":{"_cell_guid":"46f096eb-af40-44a6-9793-31a461f8bc1f","_uuid":"23354e8cc4ff792955a9a07a9b29df3b1597feb9","collapsed":true},"outputs":[],"source":["num_neighbors = 3\n","train_dates = train.transactiondate.unique()[1:]\n","# creating some columns that we will be populating\n","for n in range(1, num_neighbors+1):\n","        train['nn{}_logerror'.format(n)] = 0\n","        train['nn{}_distance'.format(n)] = 0\n","        train['nn{}_taxvaluedollarcnt'.format(n)] = 0\n","        train['nn{}_structuretaxvaluedollarcnt'.format(n)] = 0\n","        \n"]},{"cell_type":"code","execution_count":null,"metadata":{"_cell_guid":"3bb7c6fe-40aa-4d60-a979-f2275a726233","_uuid":"a3ed7578d1aa0690eb7f840907239faa4f157545","collapsed":true},"outputs":[],"source":["for d in train_dates:\n","    nbrs = NearestNeighbors(n_neighbors=num_neighbors, algorithm='ball_tree')\\\n","            .fit(train.loc[train.transactiondate<d, ['longitude','latitude']])\n","    distances, indices = nbrs.kneighbors(train.loc[train.transactiondate==d, ['longitude','latitude']])\n","    indices = indices.reshape(-1, 1)\n","    \n","    nearest_logerrors = train.loc[indices.squeeze(), 'logerror']\n","    nearest_logerrors = nearest_logerrors.values.reshape(-1, num_neighbors)\n","    \n","    nearest_taxvaluedollarcnt = train.loc[indices.squeeze(), 'taxvaluedollarcnt']\n","    nearest_taxvaluedollarcnt = nearest_taxvaluedollarcnt.values.reshape(-1, num_neighbors)\n","    \n","    nearest_structuretaxvaluedollarcnt = train.loc[indices.squeeze(), 'structuretaxvaluedollarcnt']\n","    nearest_structuretaxvaluedollarcnt = nearest_structuretaxvaluedollarcnt.values.reshape(-1, num_neighbors)\n","    \n","    train.loc[train.transactiondate==d, ['nn1_logerror', 'nn2_logerror', 'nn3_logerror']] = nearest_logerrors\n","    train.loc[train.transactiondate==d, ['nn1_distance', 'nn2_distance', 'nn3_distance']] = distances\n","    train.loc[train.transactiondate==d, ['nn1_taxvaluedollarcnt', 'nn2_taxvaluedollarcnt', 'nn3_taxvaluedollarcnt']] = nearest_taxvaluedollarcnt\n","    train.loc[train.transactiondate==d, ['nn1_structuretaxvaluedollarcnt', 'nn2_structuretaxvaluedollarcnt', 'nn3_structuretaxvaluedollarcnt']] = nearest_structuretaxvaluedollarcnt\n","    \n"]},{"cell_type":"code","execution_count":null,"metadata":{},"outputs":[],"source":["b = train_dates"]},{"cell_type":"code","execution_count":null,"metadata":{},"outputs":[],"source":["train = b"]},{"cell_type":"code","execution_count":null,"metadata":{"_cell_guid":"2819cbfa-77fd-4550-9432-538a0e503838","_uuid":"9d164f2e49601016049fe1b2cc72e36dec97a63c"},"outputs":[],"source":["to_look = 10\n","fig,ax = plt.subplots(to_look, 1, figsize=(15, 15), sharex=True)\n","for i in range(to_look):\n","    sns.distplot(train.loc[train.regionidzip == train.regionidzip.value_counts().index[i], 'logerror'],\n","                 ax=ax[i], bins=80)\n","    ax[i].set_ylabel(train.regionidzip.value_counts().index[i])"]},{"cell_type":"code","execution_count":null,"metadata":{"_cell_guid":"514818d1-ce66-4928-a3a8-0ae8b6e24cd7","_uuid":"5bee2a943ee240e5777db29c37475c449c69cb63"},"outputs":[],"source":["to_look = 10\n","fig,ax = plt.subplots(to_look, 1, figsize=(15, 15), sharex=True)\n","for i in range(to_look):\n","    sns.distplot(train.loc[train.propertyzoningdesc == train.propertyzoningdesc.value_counts().index[i], 'logerror'],\n","                 ax=ax[i], bins=80)\n","    ax[i].set_ylabel(train.propertyzoningdesc.value_counts().index[i])"]}],"metadata":{"kernelspec":{"display_name":"Python 3.10.0 64-bit","language":"python","name":"python3"},"language_info":{"codemirror_mode":{"name":"ipython","version":3},"file_extension":".py","mimetype":"text/x-python","name":"python","nbconvert_exporter":"python","pygments_lexer":"ipython3","version":"3.10.0"},"vscode":{"interpreter":{"hash":"2469a70536e4d2335a2ea8907942d0699c37342a371ac185bdb5b0aa6f073890"}}},"nbformat":4,"nbformat_minor":1}
