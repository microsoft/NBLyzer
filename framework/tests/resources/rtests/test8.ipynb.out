[{"cell_id":4,"errors":[{"line":1,"label":"b", "error_type":"ErrorType.CRITICAL", "message":"Variable uses outdated values."}],"path":[0, 2, 4]},{"cell_id":5,"errors":[{"line":1,"label":"train", "error_type":"ErrorType.CRITICAL", "message":"Variable uses outdated values."}],"path":[0, 2, 4, 5]},{"cell_id":3,"errors":[{"line":2,"label":"nbrs", "error_type":"ErrorType.TERMINAL", "message":"Idle cell."},{"line":4,"label":"distances", "error_type":"ErrorType.TERMINAL", "message":"Idle cell."},{"line":4,"label":"indices", "error_type":"ErrorType.TERMINAL", "message":"Idle cell."},{"line":7,"label":"nearest_logerrors", "error_type":"ErrorType.TERMINAL", "message":"Idle cell."},{"line":10,"label":"nearest_taxvaluedollarcnt", "error_type":"ErrorType.TERMINAL", "message":"Idle cell."},{"line":13,"label":"nearest_structuretaxvaluedollarcnt", "error_type":"ErrorType.TERMINAL", "message":"Idle cell."}],"path":[3]},{"cell_id":6,"errors":[{"line":1,"label":"to_look", "error_type":"ErrorType.TERMINAL", "message":"Idle cell."},{"line":2,"label":"fig", "error_type":"ErrorType.TERMINAL", "message":"Idle cell."},{"line":2,"label":"ax", "error_type":"ErrorType.TERMINAL", "message":"Idle cell."}],"path":[6]},{"cell_id":7,"errors":[{"line":1,"label":"to_look", "error_type":"ErrorType.TERMINAL", "message":"Idle cell."},{"line":2,"label":"fig", "error_type":"ErrorType.TERMINAL", "message":"Idle cell."},{"line":2,"label":"ax", "error_type":"ErrorType.TERMINAL", "message":"Idle cell."}],"path":[7]}]